### PR TITLE
Add json schemas for consensus stage 2

### PR DESCRIPTION
**Pull request by arnauds5**
- Add Prepare/Promise/Propose/Accept json schemas and examples/tests
- Modify the Learn schema/example for stage 2 (incompatible with stage 1)

### CODE REVIEW

**Review by Anghul — APPROVED** [review]
Good job, I will correct the go tests once this PR has been merged

File: protocol/examples/messageData/consensus_accept/wrong_accept_negative_accepted_try.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_accept/wrong_accept_negative_accepted_try.json:1]: why do we need this?

File: protocol/examples/messageData/consensus_accept/wrong_accept_negative_created_at.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_accept/wrong_accept_negative_created_at.json:1]: what is the purpose of wrong accept?

File: protocol/examples/messageData/consensus_prepare/wrong_prepare_negative_created_at.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_prepare/wrong_prepare_negative_created_at.json:1]: why do we need this?

File: protocol/examples/messageData/consensus_prepare/wrong_prepare_negative_proposed_try.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_prepare/wrong_prepare_negative_proposed_try.json:1]: why do we need this?

File: protocol/examples/messageData/consensus_promise/wrong_promise_negative_accepted_try.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_promise/wrong_promise_negative_accepted_try.json:1]: why do we need this?

File: protocol/examples/messageData/consensus_promise/wrong_promise_negative_created_at.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_promise/wrong_promise_negative_created_at.json:1]: why do we need this/

File: protocol/examples/messageData/consensus_promise/wrong_promise_negative_proposed_try.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_promise/wrong_promise_negative_proposed_try.json:1]: why do we need this?

File: protocol/examples/messageData/consensus_propose/wrong_propose_negative_created_at.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_propose/wrong_propose_negative_created_at.json:1]: why do we need this?

File: protocol/examples/messageData/consensus_propose/wrong_propose_negative_proposed_try.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (PasinduTennage) [inline protocol/examples/messageData/consensus_propose/wrong_propose_negative_proposed_try.json:1]: why do we need this?

File: protocol/examples/messageData/consensus_prepare/wrong_prepare_negative_created_at.json (hunk 3 above)
Comment (arnauds5) [reply protocol/examples/messageData/consensus_prepare/wrong_prepare_negative_created_at.json:1]: Others json schemas have some basic examples of invalid messages such as having a negative timestamp or missing fields.
Some invalid examples were asked last time : https://github.com/dedis/student_21_pop/pull/508#pullrequestreview-789180263

**Pull request merged**